src/Compilers/CSharp/Test/Syntax/Parsing/PartialEventsAndConstructorsParsingTests.cs

```diff
@@ -113,11 +122,36 @@ public void Event_Definition()
     }
 
     [Fact]
-    public void Event_Definition_Multiple()
+    public void Event_Definition_CSharp13()
+    {
+        UsingDeclaration("""
+            partial event Action E;
```

Comment:
I suspect we are in the same position we were in with records, where we can't really prompt for LangVersion with forms like `record Rec() { }` in old language versions.

I don't recall receiving much in the way of complaints about that, just something to be conscious of.

Reply:
You might do it by reporting a special error when a lookup fails for ‘record’ or ‘partial’ when binding a return type.

Reply:
That would work for constructors, not events. I'm also not sure how the diagnostic would work exactly - if it was warning or error, it would be a breaking change even for older langversions. I guess that might be fine, it seems similar to the `field` breaking change diagnostics.

Reply:
> if it was warning or error, it would be a breaking change even for older langversions.

The scenario I am describing is one where an error already occurs, for example, `partial` is used in a type position, but no such type is in scope.

> That would work for constructors, not events

I think events would work by simply parsing the partial modifier for them in any LangVersion, then reporting an error on the declaration.

But, to reiterate, I don't think there is a strong need to do these, it's probably something to only do if it is found to be convenient.

Reply:
> The scenario I am describing is one where an error already occurs, for example, partial is used in a type position, but no such type is in scope.

I see, thanks. I'm going to investigate that possibility in a future PR.

> I think events would work by simply parsing the partial modifier for them in any LangVersion, then reporting an error on the declaration.

Yes, you're right, that's what actually happens in the next PR I have prepared.